sorts/quick_sort_3_partition.py

```diff
@@ -1,25 +1,63 @@
-def quick_sort_3partition(sorting, left, right):
+"""
+This is a pure Python implementation of the three partition quick sort algorithm
+
+This algorithm splits at each iteration a mutable ordered collection in 3 parts
+1 : elements smaller than the pivot
+2 : elements equal to the pivot (not happening in 'simple' quick sort)
+3 : elements greater than the pivot
+
+For doctests run following command:
+python -m doctest -v quick_sort_3partition.py
+or
+python3 -m doctest -v quick_sort_3partition.py
+
+For manual testing run:
+python quick_sort_3partition.py
+"""
+
+
+def quick_sort_3partition(my_list: list, left: int = 0, right: int = None) -> list:
+    """
+    Implementation of quick sort algorithm, 3-Way variation.
+
+    Examples:
+    >>> quick_sort_3partition([0, 5, 3, 2, 2])
+    [0, 2, 2, 3, 5]
+
+    >>> quick_sort_3partition([])
+    []
+
+    >>> quick_sort_3partition([-2, -5, -45])
+    [-45, -5, -2]
+    """
+    if right is None:
+        right = len(my_list) - 1
     if right <= left:
-        return
-    a = i = left
-    b = right
-    pivot = sorting[left]
-    while i <= b:
-        if sorting[i] < pivot:
-            sorting[a], sorting[i] = sorting[i], sorting[a]
-            a += 1
+        return my_list
+    i: int = left
```

Comment:
left is already declared to be an int so type checkers are smart enough to know that I is an int.